src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicConfiguration.Cache.cs

```diff
@@ -47,6 +49,8 @@ private static bool GetConfigurationCacheEnabled()
         public readonly List<SslApplicationProtocol> ApplicationProtocols;
         public readonly QUIC_ALLOWED_CIPHER_SUITE_FLAGS AllowedCipherSuites;
 
+        private readonly ReadOnlySpan<byte> SettingsSpan => MemoryMarshal.AsBytes(new ReadOnlySpan<QUIC_SETTINGS>(ref Unsafe.AsRef(in Settings)));
```

Comment:
You should be able to do `MemoryMarshal.AsBytes(new ReadOnlySpan<QUIC_SETTINGS>(in Settings))` if you're on 7+

Reply:
You should be able to do `MemoryMarshal.AsBytes(new ReadOnlySpan<QUIC_SETTINGS>(in Settings))` if you're on 7+ (the docs lie and say `ref` as opposed to `ref readonly`)

Reply:
```
error CS8347: Cannot use a result of 'ReadOnlySpan<QUIC_SETTINGS>.ReadOnlySpan(ref readonly QUIC_SETTINGS)' in this context because it may expose variables referenced by parameter 'reference' outside of their declaration scope
```

It works if I inline it though, good to know.

Reply:
Ah yes, to make it safe, you need to mark the member as `[UnscopedRef]`